tests/test_compiler.py

```diff
@@ -206,3 +207,206 @@ def circuit(x: float):
         mlir_str = str(circuit.mlir)
         result_header = "func.func private @circuit(%arg0: tensor<f64>) -> tensor<f64>"
         assert result_header in mlir_str
+
+
+class TestCatalystControlFlow:
+    """Test ``qml.qjit`` with Catalyst's control-flow operations"""
+
+    def test_alternating_while_loop(self):
+        """Test simple while loop."""
+        dev = qml.device("lightning.qubit", wires=1)
+
+        @qml.qjit
+        @qml.qnode(dev)
+        def circuit(n):
+            @qml.while_loop(lambda v: v[0] < v[1])
+            def loop(v):
+                qml.PauliX(wires=0)
+                return v[0] + 1, v[1]
+
+            loop((0, n))
+            return qml.expval(qml.PauliZ(0))
+
+        assert jnp.allclose(circuit(1), -1.0)
+
+    def test_nested_while_loops(self):
+        """Test nested while loops."""
+        dev = qml.device("lightning.qubit", wires=1)
+
+        @qml.qjit
+        @qml.qnode(dev)
+        def circuit(n, m):
+            @qml.while_loop(lambda i, _: i < n)
+            def outer(i, sum):
+                @qml.while_loop(lambda j: j < m)
+                def inner(j):
+                    return j + 1
+
+                return i + 1, sum + inner(0)
```

Comment:
nitpick: `sum` is a keyword in python. why it even allows this is beyond me

Reply:
yeah, but it is not a reserved word. So, you can use it within your programs for variables.

Reply:
ah, my bad. it's not a keyword at all (docs kinda suggest keyword = reserved word in python), but it's a built-in. I'd still prefer not to override it but I suppose it's fine.